Core/include/Acts/Vertexing/AdaptiveGridTrackDensity.hpp

```diff
@@ -52,32 +44,45 @@ class AdaptiveGridTrackDensity {
 
   /// The configuration struct
   struct Config {
+    /// @param spatialTrkGridSize_ Number of bins per track in z direction
     /// @param spatialBinExtent_ The spatial extent of a bin in mm
-    Config(float spatialBinExtent_) : spatialBinExtent(spatialBinExtent_) {
-      if constexpr (temporalTrkGridSize > 1) {
-        throw std::invalid_argument(
-            "temporalBinExtent must be provided if temporalTrkGridSize > 1 "
-            "(i.e., if time vertex seeding is enabled).");
+    Config(unsigned int spatialTrkGridSize_, float spatialBinExtent_)
+        : spatialTrkGridSize(spatialTrkGridSize_),
+          spatialBinExtent(spatialBinExtent_) {
+      if (!bool(spatialTrkGridSize % 2)) {
+        throw std::invalid_argument("Please choose an odd spatialTrkGridSize.");
```

Comment:
This and the other constructor are the only ways to construct this object. This means that de-facto the defaults specify are never applied.

Now I see you use the constructors to validate the config consistency, but maybe it would be an alternative to do this in  the constructor of `AdaptiveGridTrackDensity` itself, so that `Config`  can be default constructed without having to re-list the properties in the constructors?